tests/CMakeLists.txt

```diff
@@ -164,36 +164,27 @@ target_include_directories(gsl_tests_config SYSTEM INTERFACE
     googletest/googletest/include
 )
 
-set_property(TARGET  PROPERTY FOLDER "GSL_tests")
-
-function(add_gsl_test name)
-    add_executable(${name} ${name}.cpp)
-    target_link_libraries(${name}
-        Microsoft.GSL::GSL
-        gsl_tests_config
-        ${GTestMain_LIBRARIES}
-    )
-    add_test(
-        ${name}
-        ${name}
-    )
-    # group all tests under GSL_tests
-    set_property(TARGET ${name} PROPERTY FOLDER "GSL_tests")
-endfunction()
-
-add_gsl_test(span_tests)
-add_gsl_test(span_ext_tests)
-add_gsl_test(span_compatibility_tests)
-add_gsl_test(string_span_tests)
-add_gsl_test(at_tests)
-add_gsl_test(notnull_tests)
-add_gsl_test(assertion_tests)
-add_gsl_test(utils_tests)
-add_gsl_test(owner_tests)
-add_gsl_test(byte_tests)
-add_gsl_test(algorithm_tests)
-add_gsl_test(strict_notnull_tests)
+add_executable(gsl_tests
```

Comment:
Please change this section back to the old method. We prefer each test to be compiled into its own executable rather than all tests into a single executable.

Reply:
In my tests, I [observed a failure](https://github.com/tusharpm/GSL/runs/2535279069?check_suite_focus=true) in `owner_tests` with the same error. We have two options to fix this:
1. add the delay in each process: this can be done with a modified `gtest_main.cc` including the workaround.
2. link the tests to increase the process time.

Reply:
@JordanMaples I would like to know which of the two options are preferred? With the first one, the delay in all executables adds up.
I chose the second because GTest supports filtering test cases using command line arguments.

Reply:
Sorry about the delay. Your current approach is fine. I was hesitant about this approach originally because it changed the behavior of the individual test readout in the Azure Pipeline. My worry was that consolidating the tests into a single executable would make it more difficult to debug failures. However, this is not an issue because the CTest step is decorated w/ `--output-on-failure` so there shouldn't be any difference in quickly identifying failing tests. 

Would you mind removing the function `add_gsl_test_noexcept` below similar to what you did for `add_gsl_test` so the two tests are consistent?

Reply:
Done.